modules/javafx.media/src/main/native/gstreamer/plugins/mfwrapper/mfwrapper.cpp

```diff
@@ -672,16 +642,12 @@ static gboolean mfwrapper_convert_output(GstMFWrapper *decoder)
     if (SUCCEEDED(hr))
         hr = decoder->pColorConvert->GetOutputStatus(&dwFlags);
 
-    if (dwFlags != MFT_OUTPUT_STATUS_SAMPLE_READY)
+    if (SUCCEEDED(hr) && dwFlags != MFT_OUTPUT_STATUS_SAMPLE_READY)
         return FALSE;
```

Comment:
If the previous call failed, should you really continue with the following call to `ProcessOutput`?

Reply:
Yes, if call GetOutputStatus() fails we need to call ProcessOutput(). It is based on documentation. See https://docs.microsoft.com/en-us/windows/win32/medfound/basic-mft-processing-model#process-data